pyext/src/exhaust.py

```diff
@@ -218,16 +218,20 @@ def main():
 
     print("Size of conformation matrix", conforms.shape)
 
-    if not args.skip_sampling_precision:
-        # get_rmsds_matrix modifies conforms, so save it to a file and restore
-        # afterwards (so that we retain the original IMP orientation)
-        numpy.save("conforms", conforms)
+    # get_rmsds_matrix modifies conforms, so save it to a file and restore
+    # afterwards (so that we retain the original IMP orientation)
+    numpy.save("conforms", conforms)
+
+    if not os.path.isfile("Distances_Matrix.data.npy"):
```

Comment:
What happens if the user
 - runs a simulation
 - does analysis, creating `Distances_Matrix.data.npy`
 - goes back and runs a new simulation
 - does a 2nd round of analysis?

Won't this file then be out of date and result in a hard-to-diagnose issue (matrix is for the first run but is used in the second run)? If so, one fix might be to `stat` the matrix/RMF/PDB files and only skip the matrix creation if the `.npy` file is newer than all RMF/PDB. Another would be to never reuse your cached matrix unless the user explicitly requests it with a `--use-cache` option or similar. Always better to err on the side of caution with caching.